src/app/modules/filter-bar/filter-bar.component.spec.ts

```diff
@@ -58,4 +71,93 @@ describe('FilterBarComponent', () => {
     expect(hostComponent.filterState.tcl).toBeFalsy();
     expect(hostComponent.filterState.aml).toBeFalsy();
   }));
+
+  ['tcl', 'tsl', 'aml'].forEach((type: 'tcl' | 'tsl' | 'aml') =>
+  it(`never shows ${type} validation markers when ${type} files are not being filtered out`, fakeAsync(() => {
+    // given
+    hostComponent.filterState = { tsl: true, tcl: true, aml: true };
```

Comment:
should work for filterState = { tsl: false, tcl: false, aml: false }, too, right?

Reply:
Actually, the test setup is misleading, because the method under test is not directly dependent on the filter state. The host component's `getFilteredOutMarkers` will behave in the desired way, given this state, but that is unintentionally obscuring the test setup. Arguably, the host component acts like a mock here. So, for this test, I now simply, but explicitly, set the `getFilteredOutMarkers` method to return an empty marker summary.